SurfaceTopography/NonuniformLineScan.py

```diff
@@ -132,41 +164,91 @@ def squeeze(self):
 
 
 class ScaledNonuniformTopography(DecoratedNonuniformTopography):
-    """ used when geometries are scaled
-    """
+    """Scale heights, positions, or both."""
 
-    def __init__(self, topography, scale_factor, info={}):
+    def __init__(self, topography, height_scale_factor=None, position_scale_factor=None, unit=None, info={}):
         """
-        Keyword Arguments:
-        topography  -- SurfaceTopography to scale
-        coeff -- Scaling factor
+        This topography wraps a parent topography and rescales x and z
+        coordinates according to certain rules.
+
+        Arguments
+        ---------
+        topography : :obj:`NonuniformTopographyInterface`
+            Parent topography
+        height_scale_factor : float
+            Factor to scale heights with. If None, scale factor will be
+            autodetermined from the given unit. (Default: None)
+        position_scale_factor : float
+            Factor to scale lateral positions (`physical_sizes`, etc.) with.
+            If None, scale factor will be autodetermined from the given unit.
+            (Default: None)
+        unit : str
```

Comment:
What if only one scale factor is given and the other not, with or without a given `unit`?

Reply:
The default is (or rather should be) unity for the scale factors.

Reply:
I just saw that this is actually explained in the unit docstring. Is this sufficiently clear?

Reply:
I also saw it, but it leaves some cases open. 
There are 2³=8 cases for providing arguments which we need to handle here (3x None or not None)
in the code, either by throwing an exception or taking the right values. Please see the attached drawing:
![cases](https://user-images.githubusercontent.com/9113541/123064467-9509a500-d40e-11eb-89a9-058b760deb55.jpg)

It seems to me that if not both scale factors are given and no unit is given, an exception should be raised (first three cases). If both factors are given, the unit is not needed (4th case).
If only the unit is given, the scale factors can be determined from that unit (5th case).

My question refers to the green question marks for the last three cases.
Which factors are taken if not both scale factors are given and unit is given?

Reply:
I think the best solution is to move the unit conversion to a separate pipeline function `to_unit` and simply patch through unit in the `scale` pipeline function. Then it is clear what's going on.